src/dmd/parse.d

```diff
@@ -6751,6 +6751,22 @@ final class Parser(AST) : Lexer
             error(e.loc, "`%s` must be surrounded by parentheses when next to operator `%s`", e.toChars(), Token.toChars(value));
     }
 
+    private void checkParensNot(AST.Expression left, AST.Expression right, AST.Expression exp)
+    {
+        if (left.op == TOK.not && !left.parens)
+        {
+            import dmd.expression : UnaExp;
```

Comment:
That import shouldn't be here. Either call `toChars` directly without the cast, or (perhaps, not tested) cast to `AST.UnaExp`.

Reply:
Calling `toChars` directly on `left` also outputs the `!` operator (which I need to get rid of when printing `!(a & b)`). However, casting to `AST.UnaExp` works just fine.

Reply:
Then strip the TOKnot before calling toChars?

Reply:
> Then strip the TOKnot before calling toChars?

Could you help me with that? I can't find a way to do it without casting.